ultra/ultra/adapters/default_reward_adapter.py

```diff
@@ -0,0 +1,148 @@
+# MIT License
+#
+# Copyright (C) 2021. Huawei Technologies Co., Ltd. All rights reserved.
+#
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in
+# all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+import math
+
+import numpy as np
+from scipy.spatial import distance
+
+from smarts.core.agent_interface import NeighborhoodVehicles, Waypoints
+from smarts.core.sensors import Observation
+from ultra.utils.common import ego_social_safety, get_closest_waypoint, get_path_to_goal
+
+
+_WAYPOINTS = 20  # Number of waypoints on the path ahead of the ego vehicle.
+_RADIUS = 200.0  # Locate all social vehicles within this radius of the ego vehicle.
+
+
+# This adapter requires SMARTS to pass the next _WAYPOINTS waypoints and all
+# neighborhood vehicles within a radius of _RADIUS meters in the agent's observation.
+required_interface = {
+    "waypoints": Waypoints(lookahead=_WAYPOINTS),
+    "neighborhood_vehicles": NeighborhoodVehicles(radius=_RADIUS),
+}
+
```

Comment:
It seems like _RADIUS / required_interface isn't used in this file?

Reply:
Ahh, great catch 👍. Will remove.

Reply:
Just wondering why these parameters were excluded from the `rewards` summation

https://github.com/huawei-noah/SMARTS/blob/6916ff32c37a737145488db67a612fbd2a604acc/ultra/ultra/adapters/default_reward_adapter.py#L141-L145

Reply:
Yeah good question. They were commented out when we inherited ULTRA. I think they were just testing out which elements of the reward should be kept.

Reply:
See #902 for a potential solution to get rid of `_RADIUS` and `NeighborhoodVehicles` from the default reward adapter.

Reply:
It would be best to not force these restrictions on the agents, because they won't be used in some cases (e.g. with purely image observations).

As such, maybe we can remove this as a component of the reward? I removed most of these components in the simplify-reward branch and it definitely didn't hurt.  Instead, we can have it in the info adapter for extra info.

Reply:
Yes, this would be ideal if we could remove these restrictions, and it would be great if a default info adapter could be used.

However, I have outlined in the PR (#902) that because the ULTRA episodes require specific info to record the data, if the agent wants to use these episodes, then they must use this info adapter. And if they use this info adapter, they are forced to use `Waypoints(lookahead=20)` and `NeighborhoodVehicles(radius=200.0)`. That is, they cannot then use an observation adapter with something like `Waypoints(lookahead=40)` or a reward adapter with something like `NeighborhoodVehicles(radius=50.0)`.

But after thinking about this again, perhaps this isn't all that bad. We can perhaps just ensure to set these as the default values for all default adapters and ensure to document this constraint that arises when using the default info adapter.

Reply:
Completed with the merge of #902.